src/rgw/rgw_multi.cc

```diff
@@ -253,7 +253,8 @@ int abort_multipart_upload(rgw::sal::RGWRadosStore *store, CephContext *cct,
   } while (truncated);
 
   /* use upload id as tag and do it asynchronously */
-  ret = store->getRados()->send_chain_to_gc(chain, mp_obj.get_upload_id(), false);
+  ret = store->getRados()->send_chain_to_gc(chain, mp_obj.get_upload_id());
+  // XXX: should detect ENOSPC and delete inline
```

Comment:
could you please add this part? you should be able to factor out the inline delete logic from `Object::complete_atomic_modification()` into a function for use here too

Reply:
ok